### ISSUE TITLE
Tests fail on OSX due to "_conda_pack_activate:.:38: no such file or directory: deactivate`

### ISSUE THREAD

**Issue by xhochy** [conda/conda-pack#76]
The tests fail on OSX for me. I'm running them in `zsh` with `py.test conda_pack/tests`. The failures are all because conda hasn't been loaded in my test setting and thus `deactivate` from main conda is not available.

Example error is:

```
>           assert out == 'Done\n'
E           AssertionError: assert '_conda_pack_...ivate\nDone\n' == 'Done\n'
E             - _conda_pack_activate:.:38: no such file or directory: deactivate
E               Done

conda_pack/tests/test_core.py:281: AssertionError
```

Outside of tests `conda-pack`, is usable though.

**Comment by jcrist** [comment]
Looks like the failure is due to this line printing out a deprecation warning (or on your machine erroring, perhaps new versions of conda have dropped the `deactivate` file entirely).

https://github.com/conda/conda-pack/blob/ee7efc387426f4dbe130446b4d65fca7e2a3372d/conda_pack/scripts/posix/activate#L39

This is due to the tests not being updated after the switch from `source activate` to `conda activate`. The offending line is only run if an existing conda environment is already activated (which is why it runs in the tests but not in use).

A few options:

- Use shell logic to source the deactivate script if it exists (possibly capturing stdout/stderr to avoid the deprecation warning)
- Drop compatibility with older versions of conda, use `conda deactivate` here. `conda-pack` must work without conda on the destination machine though, so some logic would be needed here to avoid that error as well.

Would you be interested in submitting a patch?

**Comment by xhochy** [comment]
Yes, already working on a patch (just keep running into more conda 4.7 issues).

**Comment by jcrist** [comment]
This was fixed by #77.
